File: backend/app/schemas/portfolio_analysis.py
"""Pydantic schemas for portfolio analysis insights."""
from pydantic import BaseModel


class PortfolioInsight(BaseModel):
    rule_id: str
    category: str   # portfolio_health | risk | income_returns | behaviour | options | mistakes
    severity: str   # info | warning | alert
    title: str
    message: str

File: frontend/src/api/portfolioAnalysis.ts
import apiClient, { unwrapResponse } from './client';
import type { ApiResponse } from '../types';

export interface PortfolioInsight {
  rule_id: string;
  category: string;
  severity: 'info' | 'warning' | 'alert';
  title: string;
  message: string;
}

export async function getPortfolioAnalysis(): Promise<PortfolioInsight[]> {
  const { data } = await apiClient.get<ApiResponse<PortfolioInsight[]>>('/api/portfolio/analysis');
  return unwrapResponse(data);
}
